src/components/Visit/clientselector.tsx

```diff
@@ -0,0 +1,62 @@
+import { Dispatch, SetStateAction } from 'react'
+import Select, { SingleValue } from 'react-select'
+
+import { FormFieldProps } from '@/components/Visit/formfield'
+import customStyles from '@/components/Visit/selectorstyles'
+import { useFirestore } from '@/context/Firebase/Firestore/context'
+import { SelectOption } from '@/types/types'
+
+interface ClientSelectorProps extends FormFieldProps {
+  setClient: Dispatch<
+    SetStateAction<{
+      clientId: string
+      clientName: string
+    }>
+  >
+}
+
+const ClientSelector = (props: ClientSelectorProps) => {
+  const {
+    userDoc: { contacts }
+  } = useFirestore()
+
+  const getClientList = () => {
+    return contacts.map((contact) => {
+      const client: SelectOption = {
+        label: contact.clientName,
+        value: contact.id
+      }
+      return client
+    })
+  }
+  const handleChange = (newValue: SingleValue<SelectOption>) => {
+    // fired when user selects an option or creates an option
+    if (newValue === null) return
+    props.setClient({ clientId: newValue.value, clientName: newValue.label })
+  }
+
+  const defaultValue: SelectOption = {
+    label: props.value || 'Select...',
+    value: props.value || ''
+  }
+
+  return (
+    <div className={props.className}>
+      <div className='flex flex-col'>
+        <label htmlFor={props.id} className='font-bold'>
+          <span className='text-primary'>{props.isRequired ? '*' : ''}</span>
+          {props.label}
+        </label>
+        <Select
+          onChange={handleChange}
+          options={getClientList()}
+          placeholder={props.placeholder}
+          styles={customStyles}
+          defaultValue={defaultValue}
+        />
```

Comment:
<img width="205" alt="Screenshot 2022-12-15 at 1 42 22 pm" src="https://user-images.githubusercontent.com/62389033/207781925-0c7c91ed-2fea-4deb-9b32-0d0aa72a0524.png">
the client option has a blue background

Reply:
I ran into this issue once and haven't been able to reproduce since.